File: DataTier/DatabaseAccess/src/main/java/mango/sep3/databaseaccess/Shared/UserAuth.java
package mango.sep3.databaseaccess.Shared;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name="UserAuth", schema = "locally")
public class UserAuth implements Serializable {
    @Id
    @Column(name = "username")
    private String username;

    @Column(name="hash")
    private String hash;

    @Column(name = "salt")
    private String salt;

    public UserAuth(){}

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getHash() {
        return hash;
    }

    public void setHash(String hash) {
        this.hash = hash;
    }

    public String getSalt() {
        return salt;
    }

    public void setSalt(String salt) {
        this.salt = salt;
    }
}

File: LogicTier/Shared/Models/UserAuth.cs
namespace Shared.Models;

public class UserAuth
{
    /// <summary>
    /// Username of the user
    /// </summary>
    public string Username { get; set; } = string.Empty;
    /// <summary>
    /// Generated salt value for the user
    /// </summary>
    public string Salt { get; set; } = string.Empty;
    /// <summary>
    /// Hash of the user's password
    /// </summary>
    public string HashPassword { get; set; } = string.Empty;
}